classes/db/DAO.inc.php

```diff
@@ -354,7 +354,13 @@ function convertFromDB($value, $type) {
 				break;
 			case 'object':
 			case 'array':
-				$value = unserialize($value);
+				// Check needed only for upgrade to 3.3, where values are retrieved before being encoded to JSON
+				$decodedValue = json_decode($value, true);
+				if (!is_null($decodedValue)) {
```

Comment:
Once we have everything moved to migrations we won't have to worry about this kind of stuff! But it'll help to note it in a git issue so this kind of thing can be cleared out when it's no longer needed. I've filed https://github.com/pkp/pkp-lib/issues/6250; please add a note here along the lines of:
```
// FIXME: pkp/pkp-lib#6250 Remove after 3.3.x upgrade code is removed (see also pkp/pkp-lib#6247)
```
...and also add a note on #6250 about this.